example/src/main/java/com/klarna/example/MainActivity.java

```diff
@@ -93,14 +128,86 @@ public void openKlarnaPreferences(View view) {
         Intent intent = new Intent(this, PreferencesActivity.class);
         intent.putExtra(PreferencesActivity.EXTRA_USER_TOKEN, getUserToken());
         startActivityForResult(intent, PREFERENCES_REQUEST_CODE);
+    }
 
+    public void onBuyPressed(View view) {
+        if (this.hasUserToken()) {
+            buyTicket();
+        } else {
+            openKlarnaRegistration();
+        }
     }
+    
+    private void buyTicket() {
+        class purchaseItemRunnable implements Runnable {
+            String reference;
+            OriginProof originProof;
+
+            purchaseItemRunnable(String reference, OriginProof originProof) {
+                this.reference = reference;
+                this.originProof = originProof;
+            }
+
+            @Override
+            public void run() {
+                try {
+                    performPurchaseOfItem(reference, originProof);
+                } catch (final Exception e) {
+                    Handler mainHandler = new Handler(getApplicationContext().getMainLooper());
+
+                    mainHandler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            showAlert("Error: " + e.toString());
+                            Log.e(getClass().getName(), "Error on performPurchaseOfItem", e);
+                        }
+                    });
+                }
+            }
+        }
 
-    public void onBuy(View view) {
         OriginProof originProof = new OriginProof(3600, "SEK", getUserToken(), getApplicationContext());
-        alert.setMessage(originProof.toString());
-        alert.show();
+
+        Thread thread = new Thread(new purchaseItemRunnable("TCKT0001", originProof));
+        thread.start();
+    }
+
+    private void performPurchaseOfItem(String reference, OriginProof originProof) throws IOException, JSONException {
+        HttpClient httpClient = new DefaultHttpClient();
+
+        HttpPost httpPost = new HttpPost("http://192.168.56.1:9292/pay");
```

Comment:
We've already talked about this, but this static IP will obviously not cut it for everyone.